src/commands/init.js

```diff
@@ -1,177 +1,46 @@
-  }
-  if (config.databaseSchema) {
-    contentToAddInDotenvFile += `DATABASE_SCHEMA=${config.databaseSchema}\n`;
-  }
-  if (config.databaseSSL !== undefined) {
-    contentToAddInDotenvFile += `DATABASE_SSL=${config.databaseSSL}\n`;
-  }
-  contentToAddInDotenvFile += `FOREST_AUTH_SECRET=${authSecret}\n`;
-  contentToAddInDotenvFile += `FOREST_ENV_SECRET=${config.forestEnvSecret}`;
-  return contentToAddInDotenvFile;
-}
-
-function commentExistingVariablesInAFile(fileData, config) {
-  const variablesToComment = {
-    'FOREST_AUTH_SECRET=': '# FOREST_AUTH_SECRET=',
-    'FOREST_ENV_SECRET=': '# FOREST_ENV_SECRET=',
-  };
-  if (config.applicationPort) {
-    variablesToComment['APPLICATION_PORT='] = '# APPLICATION_PORT=';
-  }
-  if (config.databaseUrl) {
-    variablesToComment['DATABASE_URL='] = '# DATABASE_URL=';
-    variablesToComment['DATABASE_SCHEMA='] = '# DATABASE_SCHEMA=';
-    variablesToComment['DATABASE_SSL='] = '# DATABASE_SSL=';
-  }
-  const variablesToCommentRegex = new RegExp(
-    Object.keys(variablesToComment).map((key) => `((?<!# )${key})`).join('|'),
-    'g',
-  );
-  return fileData.replace(variablesToCommentRegex, (match) => variablesToComment[match]);
-}
+const spinner = singletonGetter(Spinner);
 
-function amendDotenvFile(config, variablesToAdd) {
-  let newEnvFileData = variablesToAdd;
-  spinner.start({ text: SUCCESS_MESSAGE_ENV_VARIABLES_COPIED_IN_ENV_FILE });
-  const existingEnvFileData = fs.readFileSync('.env', 'utf8');
-  if (existingEnvFileData) {
-    const amendedExistingFileData = commentExistingVariablesInAFile(existingEnvFileData, config);
-    // NOTICE: We add the prefix only if the existing file was not empty.
-    newEnvFileData = amendedExistingFileData + ENV_VARIABLES_AUTO_FILLING_PREFIX + newEnvFileData;
+class InitCommand extends AbstractAuthenticatedCommand {
+  constructor(...args) {
+    super(...args);
+    this.environmentVariables = {};
   }
-  fs.writeFileSync('.env', newEnvFileData);
-  spinner.success();
-}
-
-function createDotenvFile(variablesToAdd) {
-  spinner.start({ text: SUCCESS_MESSAGE_ENV_FILE_CREATED_AND_FILLED });
-  fs.writeFileSync('.env', variablesToAdd);
-  spinner.success();
-}
-
-async function displayEnvironmentVariablesAndCopyToClipboard(variables) {
-  logger.info(SUCCESS_MESSAGE_DISPLAY_ENV_VARIABLES + chalk.black.bgCyan(variables));
-  await clipboardy.write(variables)
-    .then(() => logger.info(chalk.italic(SUCCESS_MESSAGE_ENV_VARIABLES_COPIED_TO_CLIPBOARD)))
-    .catch(() => null);
-}
 
-class InitCommand extends AbstractAuthenticatedCommand {
   async runIfAuthenticated() {
     try {
+      spinner.start({ text: 'Selecting your project' });
+      await spinner.attachToPromise(this.projectSelection());
+
       spinner.start({ text: 'Analyzing your setup' });
-      await spinner.attachToPromise(this.projectSelectionAndValidation());
+      await spinner.attachToPromise(this.projectValidation());
```

Comment:
Note: this was asked by Marc to split this step into 2.

Reply:
Fine, could you just add this new out put to exisiting tests?